### PR TITLE
HDDS-9978. Improve assertTrue assertions in OM integration tests

### PR DESCRIPTION
**Pull request by wzhallright**
## What changes were proposed in this pull request?
Improve assertTrue assertions in hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om
Assertions in the form:
```
assertTrue(<string>.contains(...))
```
do not provide any information about the actual value, if the assertion fails.
```
AssertionFailedError: expected: <true> but was: <false>
```
This can be improved by replacing them with:
```
import static org.assertj.core.api.Assertions.assertThat;
assertThat(<string>).contains(...)
```
which gives us more info in the form:
```
AssertionError: 

Expecting:
 <"actual string">
to contain:
 <"part"> 
```
Similarly, assertions about inequality relations, e.g.:
```
assertTrue(<x> > <y>)
```
can be replaced with:
```
assertThat(<x>).isGreaterThan(<y>);
```
The goal of this task is to find and replace such assertTrue assertions.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-9978
## CI
https://github.com/wzhallright/ozone/actions/runs/7295265484

### CODE REVIEW

**Comment by adoroszlai** [comment]
@wzhallright please resolve conflicts

**Comment by wzhallright** [comment]
> @wzhallright please resolve conflicts

Has resolved conflicts. Thanks for review.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerConfiguration.java [hunk 1]
```diff
@@ -44,6 +44,7 @@
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.Timeout;
 
+import static org.assertj.core.api.Assertions.assertThat;
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerConfiguration.java:47]: Heads up: `assertThat` is being added to `TestOzoneManagerConfiguration` and `TestOzoneManagerHAWithAllRunning` in #5844, too.  We'll need another round to resolve conflicts when that one is merged.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmMetrics.java [hunk 2]
```diff
@@ -324,8 +324,8 @@ public void testKeyOps() throws Exception {
       writeClient.commitKey(keyArgs, keySession.getId());
     } catch (Exception e) {
       //Expected Failure in preExecute due to not enough datanode
-      assertTrue(e.getMessage().contains("No enough datanodes to choose"),
-          e::getMessage);
+      assertThat(e.getMessage()).withFailMessage(e.getMessage())
+          .contains("No enough datanodes to choose");
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmMetrics.java:328]: nit: custom fail message is not needed

```suggestion
      assertThat(e.getMessage()).contains("No enough datanodes to choose");
```

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestKeyManagerImpl.java [hunk 3]
```diff
@@ -847,9 +848,8 @@ public void testLookupKeyWithLocation() throws IOException {
     // lookup key, random node as client
     OmKeyInfo key4 = keyManager.lookupKey(keyArgs, resolvedBucket(),
         "/d=default-drack/127.0.0.1");
-    assertTrue(
-        keyPipeline.getNodes().containsAll(key4.getLatestVersionLocations()
-            .getLocationList().get(0).getPipeline().getNodesInOrder()));
+    assertThat(keyPipeline.getNodes()).containsAll(key4.getLatestVersionLocations()
+            .getLocationList().get(0).getPipeline().getNodesInOrder());
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestKeyManagerImpl.java:852]:
```suggestion
    assertThat(keyPipeline.getNodes())
        .containsAll(key4.getLatestVersionLocations()
            .getLocationList().get(0).getPipeline().getNodesInOrder());
```

**Review by adoroszlai — COMMENTED** [review]
Thanks @wzhallright for updating the patch.  Let's wait with this a bit for other conflicting changes.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerConfiguration.java (hunk 1 above)
Comment (wzhallright) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerConfiguration.java:47]: ok, will roolback `TestOzoneManagerConfiguration` and `TestOzoneManagerHAWithAllRunning`

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFileInterfaces.java [hunk 4]
```diff
@@ -259,12 +261,9 @@ public void testOzFsReadWrite() throws IOException {
...
     // verify prefix directories and the file, do not already exist
-    assertTrue(
-        metadataManager.getKeyTable(getBucketLayout()).get(lev1key) == null);
-    assertTrue(
-        metadataManager.getKeyTable(getBucketLayout()).get(lev2key) == null);
-    assertTrue(
-        metadataManager.getKeyTable(getBucketLayout()).get(fileKey) == null);
+    assertNull(metadataManager.getKeyTable(getBucketLayout()).get(lev1key));
+    assertNull(metadataManager.getKeyTable(getBucketLayout()).get(lev2key));
+    assertNull(metadataManager.getKeyTable(getBucketLayout()).get(fileKey));
```
[the remark this answers is not in the document]
Comment (wzhallright) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFileInterfaces.java:266]: roolback this

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestAddRemoveOzoneManager.java [hunk 5]
```diff
@@ -131,10 +132,10 @@ private void assertNewOMExistsInPeerList(String nodeId) throws Exception {
...
       assertTrue(om.getOmRatisServer().doesPeerExist(nodeId), "New OM node " + nodeId
           + " not present in Peer list of OM " + om.getOMNodeId() + " RatisServer");
-      assertTrue(
-          om.getOmRatisServer().getCurrentPeersFromRaftConf().contains(nodeId),
-          "New OM node " + nodeId + " not present in " + "OM "
-              + om.getOMNodeId() + "RatisServer's RaftConf");
+      assertThat(
+          om.getOmRatisServer().getCurrentPeersFromRaftConf())
+              .withFailMessage("New OM node " + nodeId + " not present in " + "OM "
+              + om.getOMNodeId() + "RatisServer's RaftConf").contains(nodeId);
```
[the remark this answers is not in the document]
Comment (wzhallright) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestAddRemoveOzoneManager.java]: Okay, learned it and fixed

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestAddRemoveOzoneManager.java [hunk 6]
```diff
@@ -197,9 +199,8 @@ public void testBootstrap() throws Exception {
     GenericTestUtils.waitFor(() -> cluster.getOMLeader() != null, 500, 30000);
     OzoneManager omLeader = cluster.getOMLeader();
 
-    assertTrue(newOMNodeIds.contains(omLeader.getOMNodeId()),
-        "New Bootstrapped OM not elected Leader even though" +
-            " other OMs are down");
+    assertThat(newOMNodeIds).withFailMessage("New Bootstrapped OM not elected Leader even though" +
+        " other OMs are down").contains(omLeader.getOMNodeId());
```
[the remark this answers is not in the document]
Comment (wzhallright) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestAddRemoveOzoneManager.java]: fixed

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmMetrics.java (hunk 2 above)
Comment (wzhallright) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmMetrics.java]: fixed

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestKeyManagerImpl.java (hunk 3 above)
Comment (wzhallright) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestKeyManagerImpl.java]: fixed

**Comment by hemantk-12** [comment]
@wzhallright, I think you have to wait for HDDS-9922 and update https://github.com/apache/ozone/pull/5838/files?diff=unified&w=0#diff-7204a4e31ff9ef05e7839a26737265513d5aedc6ddeac72a9cbda7f263a0ddffR1826 as well.

**Comment by wzhallright** [comment]
> @wzhallright, I think you have to wait for [HDDS-9922](https://issues.apache.org/jira/browse/HDDS-9922) and update https://github.com/apache/ozone/pull/5838/files?diff=unified&w=0#diff-7204a4e31ff9ef05e7839a26737265513d5aedc6ddeac72a9cbda7f263a0ddffR1826 as well.

OK,Thanks for review @hemantk-12. I will rebase and update it after #5838 is merged.

**Comment by adoroszlai** [comment]
No need to wait/rebase, filesystem integration tests will be covered by separate sub-task later.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerPrepare.java [hunk 7]
```diff
@@ -296,7 +297,7 @@ public void testPrepareWithMultipleThreads() throws Exception {
               .anyMatch((vol) -> vol.getName().equals(volumeName)));
         } catch (ExecutionException ex) {
           Throwable cause = ex.getCause();
-          assertTrue(cause instanceof OMException);
+          assertInstanceOf(OMException.class, cause);
           assertEquals(
               NOT_SUPPORTED_OPERATION_WHEN_PREPARED,
               ((OMException) cause).getResult());
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerPrepare.java:303]: We can simplify this as:

```java
OMException cause = assertInstanceOf(OMException.class, cause);
assertEquals(NOT_SUPPORTED_OPERATION_WHEN_PREPARED, cause.getResult());
```

Comment (wzhallright) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerPrepare.java]: Thanks for review. fixed comments,please take a look when you have time.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerPrepare.java [hunk 8]
```diff
@@ -296,10 +297,8 @@ public void testPrepareWithMultipleThreads() throws Exception {
               .anyMatch((vol) -> vol.getName().equals(volumeName)));
         } catch (ExecutionException ex) {
           Throwable cause = ex.getCause();
-          assertTrue(cause instanceof OMException);
-          assertEquals(
-              NOT_SUPPORTED_OPERATION_WHEN_PREPARED,
-              ((OMException) cause).getResult());
+          OMException cause = assertInstanceOf(OMException.class, cause);
```
[the remark this answers is not in the document]
Comment (wzhallright) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerPrepare.java]: I noticed that too. I just had some issues with my local development environment, so I overlooked this error, fixed now.Thanks for suggestion.fixed now

**Review by adoroszlai — APPROVED** [review]

**Comment by adoroszlai** [comment]
Thanks @wzhallright for the improvement.

**Pull request merged**